lib/internal/test_runner/harness.js

```diff
@@ -186,31 +187,27 @@ async function startSubtest(subtest) {
   await subtest.start();
 }
 
-function test(name, options, fn) {
-  const parent = testResources.get(executionAsyncId()) || getGlobalRoot();
-  const subtest = parent.createSubtest(Test, name, options, fn);
-  return startSubtest(subtest);
-}
-
-function runInParentContext(Factory) {
+function runInParentContext(Factory, addShorthands = true) {
   function run(name, options, fn, overrides) {
```

Comment:
```suggestion
  async function run(name, options, fn, overrides) {
```
doing this instead would ensure it always returns a promise

Reply:
@ljharb change has caused invalid arguments to reject instead of throw so I will revert this